pkg/dogstatsd/server.go

```diff
@@ -45,9 +45,15 @@ var (
 	dogstatsdUnterminatedMetricErrors = expvar.Int{}
 
 	tlmProcessed = telemetry.NewCounter("dogstatsd", "processed",
-		[]string{"message_type", "state"}, "Count of service checks/events/metrics processed by dogstatsd")
-	tlmProcessedErrorTags = map[string]string{"message_type": "metrics", "state": "error"}
-	tlmProcessedOkTags    = map[string]string{"message_type": "metrics", "state": "ok"}
+		[]string{"message_type", "state", "origin"}, "Count of service checks/events/metrics processed by dogstatsd")
+	tlmProcessedErrorTags = map[string]string{"message_type": "metrics", "state": "error", "origin": ""}
+	tlmProcessedOkTags    = map[string]string{"message_type": "metrics", "state": "ok", "origin": ""}
+
+	// while we try to add the origin tag in the tlmProcessed metric, we want to
+	// avoid having it growing indefinitely, hence this safeguard to limit the
+	// size of this cache for long-running agent or environment with a lot of
+	// different container IDs.
+	maxOriginTagsCached = 200
```

Comment:
Did we mesure or have a rough idea of the impact of keeping this amount of counters ? If it's super cheap we might want to choose to just accumulate them to reduce overhead in CPU trying to remove the old ones.

Reply:
Convo in other comment.